src/bloqade/squin/cirq/emit/op.py

```diff
@@ -0,0 +1,125 @@
+import math
+
+import cirq
+import numpy as np
+from kirin.interp import MethodTable, impl
+
+from ... import op
+from .runtime import (
+    SnRuntime,
+    SpRuntime,
+    U3Runtime,
+    KronRuntime,
+    MultRuntime,
+    ScaleRuntime,
+    AdjointRuntime,
+    ControlRuntime,
+    UnitaryRuntime,
+    HermitianRuntime,
+    ProjectorRuntime,
+    OperatorRuntimeABC,
+    PauliStringRuntime,
+)
+from .emit_circuit import EmitCirq, EmitCirqFrame
+
+
+@op.dialect.register(key="emit.cirq")
+class EmitCirqOpMethods(MethodTable):
+
+    @impl(op.stmts.X)
+    @impl(op.stmts.Y)
+    @impl(op.stmts.Z)
+    @impl(op.stmts.H)
+    def hermitian(
+        self, emit: EmitCirq, frame: EmitCirqFrame, stmt: op.stmts.ConstantUnitary
+    ):
+        cirq_op = getattr(cirq, stmt.name.upper())
+        return (HermitianRuntime(cirq_op),)
```

Comment:
why does this emit a runtime object instead of cirq nodes?

Reply:
Yeah, it's not exactly great, but the problem is this:

If you have a composed operator expression such as a `Mult`, then you can't really return a single cirq node since it's essentially a list of two that need to be applied to the same qubits in sequence and added to the circuit.
However, the way this list is applied is also not always the same. For example, `Kron` applies a list of gates to *different* qubits (not in sequence).

So the only way I saw to do this is to return these "runtime" objects, which all have an `apply` method, that does the right thing and returns a set of cirq Operations that are added to the circuit in the impl for `qubit.Apply`.

Let me know if you have a better idea to deal with this, I'm not super happy with this implementation as it is.